### ISSUE TITLE
Multiple GeneratedEncodeJsons

### ISSUE THREAD

**Issue by julien-truffaut** [argonaut-io/argonaut#254]
I got the following error after upgrading to scala 2.12. The assembly plugin complains that there are multiple `argonaut/GeneratedEncodeJsons.class`. I fixed it locally with a custom merge strategy but it would be better if it wasn't required.

```
deduplicate: different file contents found in the following:
/Users/build/buildAgent2/system/sbt_ivy/cache/io.argonaut/argonaut_2.12/jars/argonaut_2.12-6.2-RC1.jar:argonaut/GeneratedEncodeJsons.class
/Users/build/buildAgent2/system/sbt_ivy/cache/io.argonaut/argonaut-monocle_2.12/jars/argonaut-monocle_2.12-6.2-RC1.jar:argonaut/GeneratedEncodeJsons.class
/Users/build/buildAgent2/system/sbt_ivy/cache/io.argonaut/argonaut-scalaz_2.12/jars/argonaut-scalaz_2.12-6.2-RC1.jar:argonaut/GeneratedEncodeJsons.class
```

To reproduce, you need a project with a dependency to argonaut-monocle

**Comment by julien-truffaut** [comment]
Should we move https://github.com/argonaut-io/argonaut/blob/master/project/build.scala#L46 from commonSetting to https://github.com/argonaut-io/argonaut/blob/master/build.sbt#L8

**Comment by seanparsons** [comment]
I'm just trying that now.

**Comment by seanparsons** [comment]
This looks better to me:

    ╭─sean@tower  ~/workspace/argonaut  ‹master*› 
    ╰─$ find . -iname "Generated*.scala"
    ./argonaut/target/scala-2.11/src_managed/main/argonaut/GeneratedDecodeJsons.scala
    ./argonaut/target/scala-2.11/src_managed/main/argonaut/GeneratedEncodeJsons.scala
    ./argonaut/target/scala-2.11/src_managed/main/argonaut/GeneratedCodecJsons.scala

**Comment by julien-truffaut** [comment]
awesome, fastest bug resolution ever :)

**Comment by seanparsons** [comment]
@julien-truffaut I was quite literally sat at my computer (on holiday at the moment) and saw the e-mail about this come in.

**Comment by coltfred** [comment]
@seanparsons Could we get a published version of 6.2 that has this fix?
